src/components/webview/MainProvider.ts

```diff
@@ -29,10 +30,36 @@ import {
 	CodemodNodeHashDigest,
 	selectCodemodArguments,
 } from '../../selectors/selectCodemodTree';
-import { isNeitherNullNorUndefined } from '../../utilities';
+import { buildTypeCodec, isNeitherNullNorUndefined } from '../../utilities';
 
 const X_INTUITA_ACCESS_TOKEN = 'X-Intuita-Access-Token'.toLocaleLowerCase();
 
+export const validateAccessToken = async (
+	accessToken: string,
+): Promise<boolean> => {
+	const result = await axios.post(
+		'https://telemetry.intuita.io/validateAccessToken',
+		{},
+		{
+			headers: {
+				[X_INTUITA_ACCESS_TOKEN]: accessToken,
+			},
+		},
+	);
+	const { data } = result;
+	const validation = buildTypeCodec({
```

Comment:
like I mentioned in the backend part, we should use the http status code instead of validating the response body. In case of 4xx errors, Axios will reject the promise automatically.

Reply:
addressed